src/Lunr/Gravity/MySQL/MySQLConnection.php

```diff
@@ -143,7 +144,7 @@ public function __construct($configuration, $logger, $mysqli)
 
         $this->set_configuration();
 
-        mysqli_report($configuration['db']['error_reporting'] ?? MYSQLI_REPORT_ERROR);
+        mysqli_report($configuration['db']['error_reporting'] ?? MYSQLI_REPORT_ERROR | MYSQLI_REPORT_STRICT);
```

Comment:
The default of `MYSQLI_ERROR_REPORT` is to specifically disable the throwing of `mysqli_sql_exception`, since none of the code is ready for it and it essentially breaks all of the error handling in the result classes.